File: PYTHON/StackQueue.py/first_occurance.py

S = input()
W = input()

first_index = -1

for i in range(len(S) - len(W) + 1):
    if S[i:i + len(W)] == W:
        first_index = i
        break 
if first_index != -1:
    result = S[:first_index] + S[first_index + len(W):]
else:
    result = S  
print(result.strip())

File: Nodejs/other/firstoccurance.js
const readline = require("readline").createInterface({
  input: process.stdin,
  output: process.stdout,
});

let list = [];

readline.on("line", (input) => {
  list.push(input);
  if (list.length === 2) {
    let S = list[0];
    let W = list[1];

    let words = S.split(" ");
    let result = "";
    let removed = false;

    for (let i = 0; i < words.length; i++) {
      if (words[i] === W && !removed) {
        removed = true;
        continue;
      }
      result += (result ? " " : "") + words[i];
    }

    console.log(result);
    readline.close();
  }
});
